scrapy/crawler.py

```diff
@@ -50,6 +50,12 @@ def __init__(self, spidercls, settings=None):
         self.spidercls.update_settings(self.settings)
 
         self.signals = SignalManager(self)
+
+        self.request_fingerprinter = create_instance(
```

Comment:
Any reason for putting it before creation of logging and stats component? I wonder if there is a use case for having access to them in fingerprinter's from_crawler / init methods.

Reply:
No, I simply wanted to make sure signals were available. I’ll move it further down.